_source/_posts/2022-06-19-kubernetes-jhipster-aws.md

```diff
@@ -0,0 +1,778 @@
+```
+
+Convert the secrets to encrypted secrets.
+
+```bash
+kubeseal --cert tls.crt --format=yaml -n demo < secrets.yml > sealed-secrets.yml
+```
+
+Delete the unencrypted secrets file.
+
+```
+rm secrets.yml
+```
+
+The `sealed-secrets.yml` file will look like this:
+
+```yaml
+apiVersion: bitnami.com/v1alpha1
+kind: SealedSecret
+metadata:
+  creationTimestamp: null
+  name: project-secrets
+  namespace: demo
+spec:
+  encryptedData:
+    ENCRYPT_KEY: AgBElusDb+KI9pHVeqWpzBAMnj4ksNXqXKoHdGTUQRzHi+0mnPxG+jm51QHz/6RoAGVRwVZbVnC+YpsjenbyYjz2q0U1yf+K1p3qYoOg8FlEQZYcbDRR/mQWhSZgIHTj5tRknZMkgGGkTCXGJTY4Dpjw...
+  template:
+    data: null
+    metadata:
+      creationTimestamp: null
+      name: project-secrets
+      namespace: demo
+
+```
+
+Deploy the sealed secrets to the cluster.
+
+```bash
+kubectl apply -n demo -f sealed-secrets.yml && kubectl get -n demo sealedsecret project-secrets
+```
+
+Update the registry configuration to use the encrypted value for the `ENCRYPT_KEY`. Replace the previous entry for `ENCRYPT_KEY` with the following.
+
+```
+env:
+  ...
+  - name: ENCRYPT_KEY
+    valueFrom:
+      secretKeyRef:
+        name: project-secrets
+        key: ENCRYPT_KEY
+```
+
+Deploy the cluster again.
+
+```bash
+./kubectl-apply.sh -f
+```
+
+Wait a few moments. Forward the registry again.
+
+```bash
+kubectl port-forward svc/jhipster-registry -n demo 8761
+```
+
+Open the registry and make sure you can log in.
+
+If that works, try forwarding the gateway again.
+
+```bash
+kubectl port-forward svc/gateway -n demo 8080
+```
+
+Open http://localhost:8080
+
+Make a product. Or a blog. Everything should still be working.
+
+**Once you're done, delete the EKS cluster. If you don't do this, you'll keep getting charged for the AWS resources.**
+
+```bash
+eksctl delete cluster --region=us-west-2 --name=okta-k8s
+```
+
+It wouldn't hurt to log into the console and make sure all of the resources were cleaned up.
+
+## AWS, Kubernetes, and Spring Boot Microservice is done!
+
+In this project you deployed a JHipster microservice to AWS EKS. You used the AWS CLI to create the Kubernetes resources, and you used `kubectl` to deploy the applications to the Kubernetes pods. You also used Okta as an OAuth 2.0 and OIDC provider to secure the microservice. Finally, you properly encrypted all of the sensitive configuration values using a combination of JHipster registry encryption, Kubernetes secrets, and `kubeseal`. 
+
+This project is based on two of Matt Raible's tutorials, which were helpful for me when I was writing it.
+
+- [Reactive Java Microservices with Spring Boot and JHipster](/blog/2021/01/20/reactive-java-microservices)
+- [Kubernetes to the Cloud with Spring Boot and JHipster](/blog/2021/06/01/kubernetes-spring-boot-jhipster)
+
+
+If you liked this post, there's a good chance you'll like similar ones:
+
+- [Introducing Spring Native for JHipster: Serverless Full-Stack Made Easy](/blog/2022/03/03/spring-native-jhipster)
```

Comment:
This post will go live today and will be very relevant https://github.com/oktadev/okta-blog/pull/1189